cirq-core/cirq/devices/insertion_noise_model.py

```diff
@@ -0,0 +1,71 @@
+# Copyright 2021 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from dataclasses import dataclass, field
+from typing import TYPE_CHECKING, Dict, List, Optional, Sequence
+
+from cirq import devices, ops
+from cirq.devices.noise_utils import (
+    OpIdentifier,
+    PHYSICAL_GATE_TAG,
+)
+
+if TYPE_CHECKING:
+    import cirq
+
+
+@dataclass
+class InsertionNoiseModel(devices.NoiseModel):
+    """Simple base noise model for inserting operations.
+
+    Operations generated by this model for a given moment are all added into a
+    single "noise moment", which is added before or after the original moment
+    based on `prepend`.
+
+    Args:
+        ops_added: a map of gate types (and optionally, qubits they act on) to
+            operations that should be added.
+        prepend: whether to add the new moment before the current one.
+        require_physical_tag: whether to only apply noise to operations tagged
+            with PHYSICAL_GATE_TAG.
+    """
+
+    ops_added: Dict[OpIdentifier, 'cirq.Operation'] = field(default_factory=dict)
+    prepend: bool = False
+    require_physical_tag: bool = True
+
+    def noisy_moment(
+        self, moment: 'cirq.Moment', system_qubits: Sequence['cirq.Qid']
+    ) -> 'cirq.OP_TREE':
```

Comment:
Nit: Docstring.

Reply:
This defines the `noisy_moment` method of the parent class and inherits its docstring. Compare with `noisy_moments` in [ConstantQubitNoiseModel](https://quantumai.google/reference/python/cirq/devices/ConstantQubitNoiseModel#noisy_moments).